python/ribasim/ribasim/model.py

```diff
@@ -148,6 +149,111 @@ def _write_tables(self, directory: FilePath) -> None:
                 input_entry.write(directory, self.modelname)
         return
 
+    def validate_model_node_types(self):
+        """
+        Checks whether all node types in the node field are valid
+        """
+
+        node_names_all = [
+            node_name
+            for (node_name, _) in inspect.getmembers(node_types, inspect.isclass)
+        ]
```

Comment:
Now that this expression is used three times, it would indeed make sense to make a method out of it.